File: app/android/java/app/src/main/java/com/khipu/browser2app/khenshinsamplejava/MainActivity.java
package com.khipu.browser2app.khenshinsamplejava;

import static com.khipu.khenshin.client.KhenshinKt.getKhenshinLauncherIntent;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Switch;
import android.widget.TextView;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.activity.result.ActivityResult;
import androidx.activity.result.ActivityResultCallback;
import androidx.activity.result.ActivityResultLauncher;
import androidx.activity.result.contract.ActivityResultContracts;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.khipu.khenshin.client.KhenshinColors;
import com.khipu.khenshin.client.KhenshinHeader;
import com.khipu.khenshin.client.KhenshinOptions;
import com.khipu.khenshin.client.KhenshinResult;
import static com.khipu.khenshin.client.KhenshinKt.KHENSHIN_RESULT_EXTRA;

public class MainActivity extends AppCompatActivity {
    private KhenshinOptions.Theme theme = KhenshinOptions.Theme.LIGHT;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
        final Button button = findViewById(R.id.button);
        final Switch themeSwitch = findViewById(R.id.themeSwitch);
        final EditText editText = findViewById(R.id.edit_text);
        final TextView themeLabel = findViewById(R.id.switchLabel);
        final TextView resultText = findViewById(R.id.resultText);

        String operationId = editText.getText().toString();
        ActivityResultLauncher khipuLauncher = registerForActivityResult(
                new ActivityResultContracts.StartActivityForResult(),
                new ActivityResultCallback() {
                    @Override
                    public void onActivityResult(Object o) {
                        Log.v("Callback", "mensaje final recibido " + o.toString());
                        KhenshinResult result = (KhenshinResult) ((ActivityResult) o).getData().getExtras().getSerializable(KHENSHIN_RESULT_EXTRA);
                        resultText.setText(result.toString());
                    }
                });
        themeSwitch.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (themeSwitch.isChecked()) {
                    themeLabel.setText("Modo claro");
                    theme = KhenshinOptions.Theme.LIGHT;
                } else {
                    themeLabel.setText("Modo oscuro");
                    theme = KhenshinOptions.Theme.DARK;
                }
            }
        });
        button.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                if(operationId == null || operationId.equals("") || operationId.equals("PAYMENT ID")) {
                    CharSequence text = "Ingrese un id de pago para iniciar";
                    int duration = Toast.LENGTH_SHORT;
                    Toast toast = Toast.makeText(getBaseContext() /* MyActivity */, text, duration);
                    toast.show();
                    return;
                }

//                String lightTopBarContainer = "#ffffff";
//                String lightOnTopBarContainer = "#3b9326";
//                String lightPrimary = "#3b9326";
//                String lightOnPrimary = "#ffffff";
//                String lightBackground = "#ffffff";
//                String lightOnBackground = "#333333";
//                String darkTopBarContainer = "#ffffff";
//                String darkOnTopBarContainer = "#3b9326";
//                String darkPrimary = "#3b9326";
//                String darkOnPrimary = "#ffffff";
//                String darkBackground = "#ffffff";
//                String darkOnBackground = "#333333";
                khipuLauncher.launch(getKhenshinLauncherIntent(
                        getBaseContext(),
                        operationId,
                        new KhenshinOptions.Builder()
                                .header(
                                        new KhenshinHeader.Builder()
                                                .headerLayoutId(R.layout.header_layout)
                                                .merchantNameId(R.id.merchant_name)
                                                .paymentMethodId(R.id.payment_method_value)
                                                .amountId(R.id.amount_value)
                                                .subjectId(R.id.subject_value)
                                                .build()
                                )
                                .topBarTitle("Khipu")
                                .theme(theme)
                                .skipExitPage(false)
                                .locale("es_CL")
//                                .colors(
//                                        new KhenshinColors.Builder()
//                                                .lightTopBarContainer(lightTopBarContainer)
//                                                .lightOnTopBarContainer(lightOnTopBarContainer)
//                                                .lightPrimary(lightPrimary)
//                                                .lightOnPrimary(lightOnPrimary)
//                                                .lightBackground(lightBackground)
//                                                .lightOnBackground(lightOnBackground)
//                                                .darkTopBarContainer(darkTopBarContainer)
//                                                .darkOnTopBarContainer(darkOnTopBarContainer)
//                                                .darkPrimary(darkPrimary)
//                                                .darkOnPrimary(darkOnPrimary)
//                                                .darkBackground(darkBackground)
//                                                .darkOnBackground(darkOnBackground)
//                                                .build()
//                                )
                                .build()
                ));
            }
        });
    }
}

File: app/android/kotlin/app/src/main/java/com/khipu/browser2app/khenshinsdkexample/MainActivity.kt
package com.khipu.browser2app.khenshinsdkexample

import android.app.Activity
import android.os.Bundle
import android.util.Log
import android.widget.Toast
import androidx.activity.ComponentActivity
import androidx.activity.compose.rememberLauncherForActivityResult
import androidx.activity.compose.setContent
import androidx.activity.result.ActivityResult
import androidx.activity.result.contract.ActivityResultContracts
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Button
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Switch
import androidx.compose.material3.Text
import androidx.compose.material3.TextField
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import com.khipu.browser2app.khenshinsdkexample.ui.theme.KhenshinSdkExampleTheme
import com.khipu.khenshin.client.KHENSHIN_RESULT_EXTRA
import com.khipu.khenshin.client.KhenshinColors
import com.khipu.khenshin.client.KhenshinOptions
import com.khipu.khenshin.client.KhenshinResult
import com.khipu.khenshin.client.getKhenshinLauncherIntent

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            KhenshinSdkExampleTheme {
                // A surface container using the 'background' color from the theme
                Surface(
                    modifier = Modifier.fillMaxSize(),
                    color = MaterialTheme.colorScheme.background
                ) {
                    paymentStart()
                }
            }
        }
    }
}

@Composable
fun paymentStart() {
    val context = LocalContext.current
    //val text = remember { mutableStateOf("wfg954bkelcu") }
    val text = remember {
        mutableStateOf("PAYMENT ID")
    }
    var lightTopBarContainer by remember{ mutableStateOf("#ffffff") }
    var lightOnTopBarContainer by remember{ mutableStateOf("#3b9326") }
    var lightPrimary by remember{ mutableStateOf("#3b9326") }
    var lightOnPrimary by remember{ mutableStateOf("#ffffff") }
    var lightBackground by remember{ mutableStateOf("#ffffff") }
    var lightOnBackground by remember{ mutableStateOf("#333333") }
    var darkTopBarContainer by remember{ mutableStateOf("#ffffff") }
    var darkOnTopBarContainer by remember{ mutableStateOf("#3b9326") }
    var darkPrimary by remember{ mutableStateOf("#3b9326") }
    var darkOnPrimary by remember{ mutableStateOf("#ffffff") }
    var darkBackground by remember{ mutableStateOf("#ffffff") }
    var darkOnBackground by remember{ mutableStateOf("#333333") }
    var resultText by remember{ mutableStateOf("") }
    var themeSelected by remember{ mutableStateOf(KhenshinOptions.Theme.LIGHT) }

    val khipuLauncher =
        rememberLauncherForActivityResult(ActivityResultContracts.StartActivityForResult()) { result: ActivityResult ->
            if (result.resultCode == Activity.RESULT_OK) {
                val intent = result.data
                val metadata = intent?.getSerializableExtra(KHENSHIN_RESULT_EXTRA) as KhenshinResult
                resultText = metadata.toString()
                Log.v("TAG", metadata.asJson())
            }
        }

    Column(modifier = Modifier.padding(16.dp)) {
        TextField(
            singleLine = true,
            value = text.value,
            onValueChange = { text.value = it },
            label = { Text("OperationID") }
        )

        Switch(
            checked = KhenshinOptions.Theme.LIGHT.equals(themeSelected),
            onCheckedChange = {
                if(it) {
                    themeSelected = KhenshinOptions.Theme.LIGHT
                } else {
                    themeSelected = KhenshinOptions.Theme.DARK
                }
            }
        )

        if(KhenshinOptions.Theme.LIGHT.equals(themeSelected)) {
            Text(text = "Modo claro")

        } else {
            Text(text = "Modo Oscuro")

        }



        Button(
            onClick = {
                if(text.value == null || text.value.equals("") || text.value.equals("PAYMENT ID")) {
                    val text = "Ingrese un id de pago para iniciar"
                    val duration = Toast.LENGTH_SHORT

                    val toast = Toast.makeText(context, text, duration) // in Activity
                    toast.show()
                } else {
                    khipuLauncher.launch(
                        getKhenshinLauncherIntent(
                            context = context,
                            operationId = text.value,
                            options = KhenshinOptions.Builder()
//                            .header(
//                                KhenshinHeader.Builder()
//                                    .headerLayoutId(processHeaderLayoutId)
//                                    .merchantNameId(merchantNameResourceId)
//                                    .paymentMethodId(paymentMethodResourceId)
//                                    .amountId(amountResourceId)
//                                    .subjectId(subjectResourceId)
//                                    .build()
//                            )
                                .topBarTitle("Khipu")
//                            .topBarImageResourceId(R.drawable.merchant_logo)
                                .skipExitPage(false)
                                .theme(themeSelected)
                                .locale("es_CL")
//                            .colors(
//                                KhenshinColors.Builder()
//                                    .lightTopBarContainer(lightTopBarContainer)
//                                    .lightOnTopBarContainer(lightOnTopBarContainer)
//                                    .lightPrimary(lightPrimary)
//                                    .lightOnPrimary(lightOnPrimary)
//                                    .lightBackground(lightBackground)
//                                    .lightOnBackground(lightOnBackground)
//                                    .darkTopBarContainer(darkTopBarContainer)
//                                    .darkOnTopBarContainer(darkOnTopBarContainer)
//                                    .darkPrimary(darkPrimary)
//                                    .darkOnPrimary(darkOnPrimary)
//                                    .darkBackground(darkBackground)
//                                    .darkOnBackground(darkOnBackground)
//                                    .build()
//                            )
                                .build()
                        )
                    )
                }
            },
            modifier = Modifier.padding(top = 8.dp)
        ) {
            Text("Transferir")
        }
        Text(text = "Resultado de la operación")
        Text(text = resultText)

    }
}

@Preview(showBackground = true)
@Composable
fun GreetingPreview() {
    KhenshinSdkExampleTheme {
        paymentStart()
    }
}